edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -207,17 +210,34 @@ def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: d
         """
         assert hasattr(request, 'session'), "The Django authentication middleware requires session middleware to be installed. Edit your MIDDLEWARE setting to insert 'django.contrib.sessions.middleware.SessionMiddleware'."  # noqa E501 line too long
 
-        use_jwt_cookie_requested = request.META.get(USE_JWT_COOKIE_HEADER)
+        # TODO: Move pseudocode to helper function and make this work.
+        if get_setting(ENABLE_FORGIVING_JWT_COOKIES):
+            # JSONWebTokenAuthentication gives precedence to a JWT in the authorization header, before
+            # using the JWT Cookie. Don't reconstitute the JWT Cookie unless it would be used.
+            # This simplifies later code that wants to rely on the JWT Cookie and isn't aware of
+            # this order of precendence.
+            # See https://github.com/Styria-Digital/django-rest-framework-jwt/blob/befefaa1ac1a57b1f207639a7ead472094093642/src/rest_framework_jwt/authentication.py#L90-L98
+            reconstitute_jwt_cookie = False
+            try:
+                authorization_header = force_str(JwtAuthentication.get_authorization_header(request))
+                JwtAuthentication.get_token_from_authorization_header(authorization_header)
+            except InvalidAuthorizationCredentials:
+                reconstitute_jwt_cookie = True
+        else:
+            reconstitute_jwt_cookie = request.META.get(USE_JWT_COOKIE_HEADER)
+
         header_payload_cookie = request.COOKIES.get(jwt_cookie_header_payload_name())
         signature_cookie = request.COOKIES.get(jwt_cookie_signature_name())
 
+        # TODO: This should be move later and should check if the JWT Cookie has been reconstituted, rather than
+        # checking `reconstitute_jwt_cookie`, since the reconstitution could fail.
         is_set_request_user_for_jwt_cookie_enabled = get_setting(ENABLE_SET_REQUEST_USER_FOR_JWT_COOKIE)
-        if use_jwt_cookie_requested and is_set_request_user_for_jwt_cookie_enabled:
+        if reconstitute_jwt_cookie and is_set_request_user_for_jwt_cookie_enabled:
             # DRF does not set request.user until process_response. This makes it available in process_view.
             # For more info, see https://github.com/jpadilla/django-rest-framework-jwt/issues/45#issuecomment-74996698
             request.user = SimpleLazyObject(lambda: _get_user_from_jwt(request, view_func))
 
-        if not use_jwt_cookie_requested:
+        if not reconstitute_jwt_cookie:
             metric_value = 'not-requested'
```

Comment:
The metric `request_jwt_cookie` should be renamed to `request_jwt_cookie_reconstituted` for clarity.  This is about making the cookie available, not actually authenticating with it.